src/common/compress/zst/decompress.c

```diff
@@ -0,0 +1,191 @@
+    bool frameDone;                                                 // Has the current frame completed?
+    bool done;                                                      // Is decompression done?
+} ZstDecompress;
+
+/***********************************************************************************************************************************
+Render as string for logging
+***********************************************************************************************************************************/
+static String *
+zstDecompressToLog(const ZstDecompress *this)
+{
+    return strNewFmt(
+        "{inputSame: %s, inputOffset: %zu, frameDone %s, done: %s}", cvtBoolToConstZ(this->inputSame), this->inputOffset,
+        cvtBoolToConstZ(this->frameDone), cvtBoolToConstZ(this->done));
+}
+
+#define FUNCTION_LOG_ZST_DECOMPRESS_TYPE                                                                                           \
+    ZstDecompress *
+#define FUNCTION_LOG_ZST_DECOMPRESS_FORMAT(value, buffer, bufferSize)                                                              \
+    FUNCTION_LOG_STRING_OBJECT_FORMAT(value, zstDecompressToLog, buffer, bufferSize)
+
+/***********************************************************************************************************************************
+Free decompression context
+***********************************************************************************************************************************/
+OBJECT_DEFINE_FREE_RESOURCE_BEGIN(ZST_DECOMPRESS, LOG, logLevelTrace)
+{
+    ZSTD_freeDStream(this->context);
+}
+OBJECT_DEFINE_FREE_RESOURCE_END(LOG);
+
+/***********************************************************************************************************************************
+Decompress data
+***********************************************************************************************************************************/
+static void
+zstDecompressProcess(THIS_VOID, const Buffer *compressed, Buffer *decompressed)
+{
+    THIS(ZstDecompress);
+
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(ZST_DECOMPRESS, this);
+        FUNCTION_LOG_PARAM(BUFFER, compressed);
+        FUNCTION_LOG_PARAM(BUFFER, decompressed);
+    FUNCTION_LOG_END();
+
+    ASSERT(this != NULL);
+    ASSERT(this->context != NULL);
+    ASSERT(decompressed != NULL);
+
+    // When there is no more input then decompression is done
+    if (compressed == NULL)
+    {
+        // If the current frame being decompressed was not completed then error
+        if (!this->frameDone)
+            THROW(FormatError, "unexpected eof in compressed data");
+
+        this->done = true;
+    }
+    else
+    {
+        // Initialize input/output buffer
+        ZSTD_inBuffer in = {.src = bufPtrConst(compressed) + this->inputOffset, .size = bufUsed(compressed) - this->inputOffset};
+        ZSTD_outBuffer out = {.dst = bufRemainsPtr(decompressed), .size = bufRemains(decompressed)};
```

Comment:
Same comment about formatting structures as before. I don't really have a preference but we should probably make a decision on the standard (currently the coding standard only discusses structure declaration not initialization).

Reply:
We have plenty of cases where we only init one field and I'd certainly hate to split those out. I'd rather leave the initialization on one line if it fits.